src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -42,6 +42,16 @@ internal sealed partial class SolutionState
     public SolutionOptionSet Options { get; }
     public IReadOnlyList<AnalyzerReference> AnalyzerReferences { get; }
 
+    /// <summary>
+    /// Fallback analyzer config options by language. The set of languages does not need to match the set of langauges of projects included in the surrent solution snapshot.
+    /// </summary>
+    public ImmutableDictionary<string, StructuredAnalyzerConfigOptions> FallbackAnalyzerOptions { get; } = ImmutableDictionary<string, StructuredAnalyzerConfigOptions>.Empty;
+
+    /// <summary>
+    /// Number of projects in the solution of the given language.
+    /// </summary>
+    internal ImmutableDictionary<string, int> ProjectCountByLanguage { get; } = ImmutableDictionary<string, int>.Empty;
```

Comment:
can this be made private?

Reply:
Needs to be accessible from Workspace type